clang/include/clang/Basic/Builtins.td

```diff
@@ -4765,6 +4765,12 @@ def HLSLAll : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "bool(...)";
 }
 
+def HLSLAnd : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_and"];
```

Comment:
I suppose the reason there is no c++ builtin existing here is because this case was covered only for scalars and further via the and operator.